math/matrix/src/TMatrixTBase.cxx

```diff
@@ -622,8 +622,8 @@ Element TMatrixTBase<Element>::Max() const
 template<class Element>
 void TMatrixTBase<Element>::Draw(Option_t *option)
 {
-   gROOT->ProcessLine(Form("THistPainter::PaintSpecialObjects((TObject*)0x%lx,\"%s\");",
-                           (ULong_t)this, option));
+   gROOT->ProcessLine(Form("THistPainter::PaintSpecialObjects((TObject*)0x%zx,\"%s\");",
+                           (size_t)this, option));
```

Comment:
As "this" is a pointer, can we not use `Longptr_t` here, too?

Reply:
Well, nope, the `%zx` formatting requires `size_t` and is fully cross architecture. And using `%lx` or `%llx` will then conflict, depending on the platform. The best and more portable way is really this one (it took me some time to find the best way to fix type casting errors and warnings)

Reply:
Right, while `std::uint64_t` is a good, conservative target type for a pointer cast, it's all but obvious which format specifier to use... Thanks!